### PR TITLE
Riegeli: implement transposed record chunk encoder

### PR DESCRIPTION
**Pull request by schroederc**
This is a rough implementation of transposed Riegeli chunks.  Instead of
doing the full state machine optimization done in C++, this
implementation uses trampoline states at discreet intervals to simplify
the encoding implementation.  Despite the state machine overhead,
compression is still significantly improved over regular chunking when
encoding protocol buffer messages.

| Encoding test                             | Encoded size   |
|-------------------------------------------|---------------:|
| TestReadWriteProto/defaults               |  55464 bytes   |
| TestReadWriteProto/transpose              |  32445 bytes   |
| TestReadWriteProto/uncompressed           | 722937 bytes   |
| TestReadWriteProto/uncompressed,transpose | 581221 bytes   |
| TestReadWriteProto/brotli                 |  55464 bytes   |
| TestReadWriteProto/brotli,transpose       |  32445 bytes   |
| TestReadWriteProto/zstd                   | 218027 bytes   |

### CODE REVIEW

File: kythe/go/util/riegeli/riegeli_test.go [hunk 1]
```diff
@@ -93,6 +113,60 @@ func testReadWriteStrings(t *testing.T, opts *WriterOptions) {
 	}
 }
 
+func writeProtos(t *testing.T, opts *WriterOptions, n int) *bytes.Buffer {
+	var buf bytes.Buffer
```
Comment (creachadair) [inline kythe/go/util/riegeli/riegeli_test.go]: ```t.Helper()```

File: kythe/go/util/riegeli/riegeli_test.go [hunk 2]
```diff
@@ -93,6 +113,60 @@ func testReadWriteStrings(t *testing.T, opts *WriterOptions) {
...
+			t.Fatalf("Error PutProto(%d): %v", i, err)
+		}
+	}
+	if err := wr.Close(); err != nil {
+		t.Fatalf("Close error: %v", err)
+	}
+	return &buf
+}
+
+func numToProto(i int) *rtpb.Complex {
```
Comment (creachadair) [inline kythe/go/util/riegeli/riegeli_test.go]: Please add a comment here explaining what this does, generally.

Comment (schroederc) [reply kythe/go/util/riegeli/riegeli_test.go]: 👍

File: kythe/go/util/riegeli/riegeli_test.go (hunk 1 above)
Comment (schroederc) [reply kythe/go/util/riegeli/riegeli_test.go]: 👍

**Pull request merged**